### DISCUSSION TITLE
Can someone please provide a full copy/paste of one of their .bat files used to run a model?

### DISCUSSION THREAD

**Discussion by viperwasp** [oobabooga/text-generation-webui#1644] [Q&A] [answered]
I reinstalled oobabooga using the new Windows one click installer. My bat files fail to load models. 
Here is the script I use. 

For example this...
********
@echo off

@echo Starting the web UI...

cd /D "%~dp0"

set MAMBA_ROOT_PREFIX=%cd%\installer_files\mamba
set INSTALL_ENV_DIR=%cd%\installer_files\env

if not exist "%MAMBA_ROOT_PREFIX%\condabin\micromamba.bat" (
  call "%MAMBA_ROOT_PREFIX%\micromamba.exe" shell hook >nul 2>&1
)
call "%MAMBA_ROOT_PREFIX%\condabin\micromamba.bat" activate "%INSTALL_ENV_DIR%" || ( echo MicroMamba hook not found. && goto end )
cd text-generation-webui

call python server.py --auto-devices --chat --model gpt4-x-alpaca-13b-native-4bit-128g --wbits 4 --groupsize 128

:end
pause
********

Error I get. 

Starting the web UI...
The system cannot find the path specified.
MicroMamba hook not found.
Press any key to continue . . .

I believe I know how to copy over MicroMamba or something but is my .bat file correct? Or is it outdated? I got it from a guide for a former install of Oobabooga. Thanks.

**Comment by jparmstr** [comment]
I reinstalled a few days ago, and it seems to be using Miniconda instead of Micromamba now. There is a webui.py script in the same folder as the .bat files that needs to be edited. Specifically the line with the comment "# put your flags here!"

I liked the old setup better where arguments could be provided via the .bat file, so you could just create as many .bat files as you had models. My workaround is to create copies of webui.py for each model, for example "webui_llava13b.py" and .bat files that refer to those copies.

**Reply by viperwasp** [reply]
> I reinstalled a few days ago, and it seems to be using Miniconda instead of Micromamba now. There is a webui.py script in the same folder as the .bat files that needs to be edited. Specifically the line with the comment "# put your flags here!"
> 
> I liked the old setup better where arguments could be provided via the .bat file, so you could just create as many .bat files as you had models. My workaround is to create copies of webui.py for each model, for example "webui_llava13b.py" and .bat files that refer to those copies.

Thank you Jparmstr no wonder what I was doing could not work. I think I understand. I found the webui file. I just copy and paste it and rename it and give each one the arguments I want. How do I make a bat file to call them? Can you please provide your .bat file example? I just want to know what you put in one of your bat files. I think I can figure it out once I see that. Thank you.

**Reply by jparmstr** [reply, marked as answer]
Sure! This is how I edited one of the webui.py files (renamed webui_llava13b.py), starting on line 144:

```
def run_model():
    os.chdir("text-generation-webui")
    run_cmd("python server.py --model llava-13b-4bit-128g --wbits 4 --group 128 --chat --model_type=llama --extensions llava  --pre_layer 29")  # put your flags here!
```

And this is the edit to the .bat file, on line 49. Everything else is exactly the same:

`call python webui_llava13b.py`

**Reply by viperwasp** [reply]
Thanks I will test this soon. With one of my models etc. I think you answered what I needed. I appreciate it jparmstr. 
It worked perfectly!
